SaveData/R/SaveData.R

```diff
@@ -97,16 +98,9 @@ SaveData <- function(df, key, outfile, logfile = NULL, appendlog = FALSE, sortby
 
       reordered_colnames <- c(key,
                               colnames(df[!colnames(df) %in% key]))
-
-      args <- list(df)
-      i <- 2
-      while(i<length(key)+2) {
-        args[[i]] <- df[[(key[[i-1]])]]
-        i <- i + 1
-      }
       
       if (sortbykey) {
-        df <- do.call(arrange, args)  # sort by key values
+        df <- setorderv(df, key)  # sort by key values
```

Comment:
`setorderv` works by reference, so I think that the assignment is unnecessary. Doing 

```R
        setorderv(df, key)  # sort by key values
```
should work.

Reply:
Thanks @santiagohermo! Implemented in f4c8cf0.